cli/src/main/java/org/projectnessie/tools/catalog/migration/cli/CatalogMigrationCLI.java

```diff
@@ -0,0 +1,507 @@
+              + "Should not be used with `--identifiers` or '--identifiers-from-file' option.")
+  private String identifiersRegEx;
+
+  @CommandLine.Option(
+      names = {"--dry-run"},
+      description =
+          "optional configuration to simulate the registration without actually registering. Can learn about a list "
+              + "of the tables that will be registered by running this.")
+  private boolean isDryRun;
+
+  @CommandLine.Option(
+      names = {"--delete-source-tables"},
+      description =
+          "optional configuration to delete the table entry from source catalog after successfully registering it "
+              + "to target catalog.")
+  private boolean deleteSourceCatalogTables;
+
+  @CommandLine.Option(
+      names = {"--output-dir"},
+      description =
+          "optional local output directory path to write CLI output files like `failed_identifiers.txt`, "
+              + "`failed_to_delete_at_source.txt`, `dry_run_identifiers.txt`. "
+              + "Uses the present working directory if not specified.")
+  private Path outputDirPath;
+
+  private boolean disablePrompts;
+
+  private static final int BATCH_SIZE = 100;
+
+  public CatalogMigrationCLI() {}
+
+  public static void main(String... args) {
+    CommandLine commandLine = new CommandLine(new CatalogMigrationCLI());
+    commandLine.setUsageHelpWidth(150);
+    int exitCode = commandLine.execute(args);
+    System.exit(exitCode);
+  }
+
+  public void disablePrompts() {
+    this.disablePrompts = true;
+  }
+
+  @Override
+  public Integer call() {
+    validateIdentifierOptions();
+
+    PrintWriter printWriter = commandSpec.commandLine().getOut();
+    Configuration sourceCatalogConf = new Configuration();
+    if (sourceHadoopConf != null && !sourceHadoopConf.isEmpty()) {
+      sourceHadoopConf.forEach(sourceCatalogConf::set);
+    }
+    Catalog sourceCatalog =
+        CatalogUtil.loadCatalog(
+            Objects.requireNonNull(catalogImpl(sourceCatalogType, sourceCustomCatalogImpl)),
+            sourceCatalogType.name(),
+            sourceCatalogProperties,
+            sourceCatalogConf);
+    printWriter.println(String.format("%nConfigured source catalog: %s", sourceCatalogType.name()));
+
+    Configuration targetCatalogConf = new Configuration();
+    if (targetHadoopConf != null && !targetHadoopConf.isEmpty()) {
+      targetHadoopConf.forEach(targetCatalogConf::set);
+    }
+    Catalog targetCatalog =
+        CatalogUtil.loadCatalog(
+            Objects.requireNonNull(catalogImpl(targetCatalogType, targetCustomCatalogImpl)),
+            targetCatalogType.name(),
+            targetCatalogProperties,
+            targetCatalogConf);
+    printWriter.println(String.format("%nConfigured target catalog: %s", targetCatalogType.name()));
+
+    List<TableIdentifier> tableIdentifiers = processIdentifiersInput(printWriter);
+
+    if (!canProceed(printWriter, sourceCatalog)) {
```

Comment:
Why pass that `printWriter` around everywhere, although it is always accessible via `commandSpec`? That's not neccessary.